pkg/operation/botanist/addons.go

```diff
@@ -430,6 +434,10 @@ func (b *Botanist) generateCoreAddonsChart() (*chartrenderer.RenderedChart, erro
 		nodeNetwork = b.Shoot.GetNodeNetwork()
 	)
 
+	if _, ok := b.Secrets["vpa-tls-certs"]; ok {
+		verticalPodAutoscaler["admissionController"].(map[string]interface{})["caCert"] = b.Secrets["vpa-tls-certs"].Data[secrets.DataKeyCertificateCA]
+	}
```

Comment:
Why don't you simply move that up to the `admissionController` section?

Reply:
if the vpa is not enabled for the shoot, the secret `vpa-tls-certs` is not deployed. So adding caCert will require this check to prevent null dereference.

Reply:
I see, makes sense.. let's keep it like this then. Something we should get rid when we refactor this deployment as part of #2754